homepage/tests.py

```diff
@@ -103,3 +104,117 @@ def test_create_multiple_courses():
     # make sure that multiple courses are saved accurately
     courses = [Course.objects.get(pk=arg[0]).get_details() for arg in details]
     assert courses == details
+
+
+# -----------prerequisites tests----------- #
+
+# three courses:
+@pytest.fixture
+def courses():
+    courses_list = [
+         Course(1, "Course1", True, 3, "N/A", 1.3, 4, 5, 1),
+         Course(2, "Course2", False, 2, "N/A", 3.5, 2, 13, 5),
+         Course(3, "Course3", False, 1, "N/A", 2, 2.5, 0, 0)]
+    for course in courses_list:
+        course.save()
+
+    return courses_list
+
+
+# creating a new prerequisite log
+@pytest.mark.django_db
+def test_create_new_prerequisite(courses):
+    flag = True
+    preq = Prerequisites(course_id=courses[0], req_course_id=courses[1], req_code=Prerequisites.Req_Code.BEFORE)
+    try:
+        preq.clean_fields()
+    except ValidationError:
+        flag = False
+
+    assert flag
+
+    preq.save()
+
+    assert Prerequisites.objects.filter(pk=preq.id).exists()
+    assert Prerequisites.objects.filter(pk=preq.id).count() == 1
+
+
+# creating a new invalid prerequisite log - req_code invalid
+@pytest.mark.django_db
+def test_create_new_invalid_prerequisite(courses):
+    flag = False
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=3)
+    try:
+        preq.clean_fields()
+    except ValidationError:
+        flag = True
+
+    assert flag
+
+
+@pytest.mark.django_db
+def test_prerequisite_print(courses, capsys):
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE)
+    preq.save()
+
+    msg = f"req_cour = Course1, desired_cour = Course2, req_code = {Prerequisites.Req_Code.BEFORE}\n"
+    print(preq)
+    assert capsys.readouterr().out == msg
+
+
+@pytest.mark.django_db
+# test for Course2 which has prerequisite - Course1
+def test_get_prerequisites_for_course_with_one_preqs(courses):
+    preq = Prerequisites(course_id=courses[1], req_course_id=courses[0], req_code=Prerequisites.Req_Code.BEFORE)
+    preq.save()
+
+    assert Prerequisites.get_prerequisites_for_course(courses[1]).exists()
+    assert Prerequisites.get_prerequisites_for_course(courses[1]).count() == 1

```

Comment:
The first assertion is redundant since the second one will fail if it does not exist, so it covers both checks.
May I suggest retrieving the list of preq' from the method and then:
```suggestion
    preq_list = Prerequisites.get_prerequisites_for_course(courses[1])
    assert preq_list[0] == preq and len(preq_list) == 1
```
That way you're checking the full object and assure that only one was returned.